src/rmp/src/Restructure.cpp

```diff
@@ -179,17 +189,47 @@ void Restructure::runABC()
     if (writeAbcScript(abc_script_file)) {
       std::string abc_command = std::string("yosys-abc < ") + abc_script_file;
       if (logfile_ != "")
-        abc_command = abc_command + " > " + logfile_;
-      int ret = std::system(abc_command.c_str());
-      if (ret) {
-        logger_->warn(
-            RMP,
-            11,
-            "ABC failed with code {}, please check messages for details.",
-            ret);
-        continue;
+        abc_command = abc_command + " > " + logfile_ + std::to_string(opt_mode);
+
+      pid_t child_pid = fork();
+      if (child_pid == 0) {
+        int ret = std::system(abc_command.c_str());
+        exit(ret);
+      }
+
+      if(child_pid > 0){
+        child_proc[opt_mode - opt_mode_init] = child_pid;
       }
+      
+      files_to_remove.emplace_back(abc_script_file);
+    }
+  }
+
+  for (int child_idx = 0; child_idx < child_proc.size(); child_idx++) {
+    pid_t child = child_proc[child_idx];
+    if (child == 0) {
+      logger_->warn(RMP, 14, "ABC ({}) failed to run.", child_idx + 1);
```

Comment:
@ahmadelrouby also point out the log file to see error/issue? Process id is not useful to user, script file and error code would be better.